configure.ac

```diff
@@ -1351,6 +1351,8 @@ HWLOC_DO_AM_CONDITIONALS
 AM_CONDITIONAL([have_hwloc], [test "${have_hwloc}" = "yes"])
 AM_CONDITIONAL([use_embedded_hwloc], [test "${use_embedded_hwloc}" = "yes"])
 
+AM_CONDITIONAL([HAVE_HWLOC], [test x$have_hwloc = xyes])
```

Comment:
This is redundant since `HAVE_HWLOC` is defined in line 1346. Please remove this.

Reply:
Without this change I was not able to use “HAVE_HWLOC” in the Makefile. I can try the lower case “have_hwloc” to see if that works.

Reply:
@yfguo I was looking at your PR #3766. So in this PR I can actually use "HYDRA_HAVE_HWLOC" (assuming that PR gets merged first). Right?

Reply:
No. `HYDRA_HAVE_HWLOC` is defined in hydra's `configure.ac` so we should not rely on that for MPID code. If `HAVE_HWLOC` is not working I need to look into this problem and fix it along with other changes in PR #3766.